ovirt_imageio/_internal/qemu_nbd.py

```diff
@@ -89,7 +90,11 @@ def url(self):
     def start(self):
         # Implement "auto" cache mode, planned for qemu 7.
         if self.cache is None:
-            self.cache = "none" if self._can_use_direct_io() else "writeback"
+            if self._can_use_direct_io() or self._is_block_dev():
+                self.cache = "writeback"
+            else:
+                self.cache = "none"
```

Comment:
This is wrong, will change self.cache=None to self.cache="writeback" for file that can
use direct I/O.

But even if we fix it to:

    if self._can_use_direct_io() or not self._is_block_dev():
        self.cache = "none"
    else:
        self.cache = "writeback"

This is still not correct since block devices typically support direct I/O
and if direct I/O is not supported opening the image using direct I/O should
fail. Vdsm was using this kind of check for 10 years and we never found a case
when this did not work. But maybe the check need to be improved, and we need
to try to read from the device instead of just opening it?

Can you share information from your setup?

On the host:

    kubectl get pvc/xxx ... -o yaml
    kubectl get pv/xxx ... -o yaml

In the container:

    udevadm info /dev/xxx
    dd if=/dev/urandom bs=1m count=1 of=/var/tmp/data
    qemu-img convert -f raw -O raw -t none /var/tmp/data /dev/xxx